### PR TITLE
Feature/multi bot goals

### PR DESCRIPTION
**Pull request by dylanwishner**
https://offworld-ai.atlassian.net/jira/software/c/projects/SWTP/boards/182?assignee=622faca98c349300687b500b&selectedIssue=SWTP-1691
https://offworld-ai.atlassian.net/jira/software/c/projects/SWTP/boards/182?assignee=622faca98c349300687b500b&selectedIssue=SWTP-1706

- Add ROS publisher for the waypoint of each robot, as well as subscriber that will send the active waypoint to the rosboard client
- Calculate the angle between the robot and the goal which will be sent as the yaw
- Goals were changed from being sent in the odom frame to the map frame. This was causing the map in the HMI to be slightly rotated since the robot's transform and map's parent frame didn't match

### CODE REVIEW

File: rosboard/handlers.py [hunk 1]
```diff
@@ -57,6 +62,11 @@ def open(self):
             "hostname": socket.gethostname(),
             "version": __version__,
         }], separators=(',', ':')))
+        
+        # Publishers for active bot waypoint
+        qos =  rclpy.qos.QoSProfile(depth=1, durability=rclpy.qos.QoSDurabilityPolicy.TRANSIENT_LOCAL)
+        self.surveyor_waypoint_pub = rospy._node.create_publisher(Float32MultiArray, "/geosurvey/active_waypoint", qos)
+        self.digger_waypoint_pub = rospy._node.create_publisher(Float32MultiArray, "/digger/active_waypoint", qos)
```
Comment (rohitnambiarOW) [inline rosboard/handlers.py:70]: Looks good for now but later down we want to make sure we can handle multiple robots later

File: rosboard/handlers.py [hunk 2]
```diff
@@ -201,17 +211,24 @@ def on_message(self, message):
 
             op_name = argv[1].get("op_name")
             args = argv[1].get("args")
+            print("Sending bot to " + str(args))
```
Comment (rohitnambiarOW) [inline rosboard/handlers.py]: Should we keep this print statement?  may want to comment it out

File: rosboard/html/js/viewers/WaypointPointCloudViewer.js [hunk 3]
```diff
@@ -214,19 +234,30 @@ class WaypointPointCloudViewer extends Viewer {
...
         this.pointsBuffer.setAttribute('position', new THREE.BufferAttribute(points, 3));
         this.pointsBuffer.rotateX(-Math.PI / 2);
     }
+
+    getBotToGoalAngle(botName) {
+        if (this.bots[botName].position.x == null) {
+            return null;
+        }
+
+        const dx = this.bots[botName].waypoint.position.x - this.bots[botName].position.x;
```
Comment (rohitnambiarOW) [inline rosboard/html/js/viewers/WaypointPointCloudViewer.js:257]: Should we check to verify waypoint is there?

**Review by rohitnambiarOW — APPROVED** [review]

File: rosboard/handlers.py (hunk 1 above)
Comment (dylanwishner) [reply rosboard/handlers.py:70]: Yeah, it also relies on there being bots of different species. Definitely needs to be revisited

File: rosboard/handlers.py (hunk 2 above)
Comment (dylanwishner) [reply rosboard/handlers.py]: Thanks, removed it. Was just used for debugging

File: rosboard/html/js/viewers/WaypointPointCloudViewer.js (hunk 3 above)
Comment (dylanwishner) [reply rosboard/html/js/viewers/WaypointPointCloudViewer.js:257]: Good call, done

**Pull request merged**